cargo-pgrx/src/command/schema.rs

```diff
@@ -376,195 +295,272 @@ pub(crate) fn generate_schema(
-    PgrxPgSysStub::from_symbols(&symbols_to_stub)?.write_to_file(&postmaster_stub_file)?;
+    command.args(profile.cargo_args());
 
-    let mut so_rustc_invocation = crate::env::rustc();
-    so_rustc_invocation.stderr(Stdio::inherit());
+    if let Some(log_level) = &log_level {
+        command.env("RUST_LOG", log_level);
+    }
 
-    if let Ok(rustc_flags_str) = std::env::var("RUSTFLAGS") {
-        let rustc_flags = rustc_flags_str.split_whitespace().collect::<Vec<_>>();
-        if !rustc_flags.is_empty() {
-            so_rustc_invocation.args(rustc_flags);
-        }
+    if !features_arg.trim().is_empty() {
+        command.arg("--features");
+        command.arg(&features_arg);
+    }
+
+    if features.no_default_features {
+        command.arg("--no-default-features");
+    }
+
+    if features.all_features {
+        command.arg("--all-features");
     }
 
-    so_rustc_invocation.args([
-        "--crate-type",
-        "cdylib",
-        "-o",
-        postmaster_stub_built
-            .to_str()
-            .ok_or(eyre!("could not call postmaster_stub_built.to_str()"))?,
-        postmaster_stub_file
-            .to_str()
-            .ok_or(eyre!("could not call postmaster_stub_file.to_str()"))?,
-    ]);
-
-    let so_rustc_invocation_str = format!("{:?}", so_rustc_invocation);
-    tracing::debug!(command = %so_rustc_invocation_str, "Running");
-    let output = so_rustc_invocation.output().wrap_err_with(|| {
-        eyre!("could not invoke `rustc` on {}", &postmaster_stub_file.display())
-    })?;
-
-    let code = output.status.code().ok_or(eyre!("could not get status code of build"))?;
-    tracing::trace!(status_code = %code, command = %so_rustc_invocation_str, "Finished");
-    if code != 0 {
-        return Err(eyre!("rustc exited with code {}", code));
+    for arg in flags.split_ascii_whitespace() {
+        command.arg(arg);
     }
 
-    std::fs::write(&postmaster_hash_file, postmaster_bin_hash)
-        .wrap_err("could not write postmaster stub hash")?;
+    command.arg("--");
 
-    Ok(postmaster_stub_built)
+    command.args(["--cfg", "pgrx_embed"]);
+
+    command.env("PGRX_EMBED", embed_path.as_ref());
+
+    let command_str = format!("{:?}", command);
+    eprintln!(
+        "{} for SQL generation with features `{}`",
+        "  Rebuilding".bold().green(),
+        features_arg,
+    );
+
+    tracing::debug!(command = %command_str, "Running");
+    let cargo_output =
+        command.output().wrap_err_with(|| format!("failed to spawn cargo: {}", command_str))?;
+    tracing::trace!(status_code = %cargo_output.status, command = %command_str, "Finished");
+
+    if !cargo_output.status.success() {
+        // We explicitly do not want to return a spantraced error here.
+        std::process::exit(1)
+    }
+
+    Ok(())
+}
+
+fn compute_sql(profile: &CargoProfile) -> eyre::Result<()> {
+    let mut bin = get_target_dir()?;
+    bin.push(profile.target_subdir());
+    bin.push("pgrx_embed");
```

Comment:
We shouldn't be placing the binary at `/target/{profile}/pgrx_embed`. This kind of intermediate artifact should be using something like `env::var("OUT_DIR")`. Otherwise, we encounter problems if we want to do any kind of build parallelism and will probably eventually need to roll our own flocking just to stop ourselves from stomping on ourselves.

Reply:
Note that the "further namespacing" by using the concatenated binary name of `ext_name_pgrx_embed` (or whatever) is sufficient.